packages/flutter_tools/lib/src/device.dart

```diff
@@ -556,6 +556,10 @@ class AndroidDevice extends Device {
           'Device $deviceID is not authorized.\n'
           'You might need to check your device for an authorization dialog.'
         );
+      } else if (offlineRegex.hasMatch(line)) {
+        Match match = offlineRegex.firstMatch(line);
```

Comment:
not that it matters here but in general hasMatch() followed by firstMatch() is duplication of potentially significant effort.